common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -36,10 +39,14 @@
  * {@link ByteBuffer}. It is often used in conjunction with {@link TextHeaders}.
  */
 public final class AsciiString extends ByteString implements CharSequence, Comparable<CharSequence> {
-
+    private static final HashCodeGenerator CASE_INSENSITIVE_HASHER =
+            PlatformDependent.hashCodeGeneratorAsciiCaseInsensitive();
     private static final char MAX_CHAR_VALUE = 255;
     public static final AsciiString EMPTY_STRING = new AsciiString("");
 
+    private String string;
+    private int caseInsensitiveHashCode = CASE_INSENSITIVE_HASHER.emptyHashValue();
```

Comment:
I think this discussion also applies to some other member variables (`string`, `ByteString.hash`).